chia/data_layer/data_layer_server.py

```diff
@@ -0,0 +1,64 @@
+import aiosqlite
+from aiohttp import web
+from chia.data_layer.data_store import DataStore
+from chia.util.db_wrapper import DBWrapper
+from chia.types.blockchain_format.tree_hash import bytes32
+from tests.core.data_layer.util import generate_big_datastore
+
+
+class DataLayerServer:
+    async def handle_tree_root(self, request: web.Request) -> web.Response:
+        tree_id = request.rel_url.query["tree_id"]
+        tree_id_bytes = bytes32.from_hexstr(tree_id)
+        tree_root = await self.data_store.get_tree_root(tree_id_bytes)
+        result = {
+            "tree_id": tree_id,
+            "generation": tree_root.generation,
+            "node_hash": str(tree_root.node_hash),
+            "status": tree_root.status.value,
+        }
+        return web.json_response(result)
+
+    async def handle_tree_nodes(self, request: web.Request) -> web.Response:
+        node_hash = request.rel_url.query["node_hash"]
+        tree_id = request.rel_url.query["tree_id"]
+        root_hash = request.rel_url.query["root_hash"]
+        node_hash_bytes = bytes32.from_hexstr(node_hash)
+        tree_id_bytes = bytes32.from_hexstr(tree_id)
+        root_hash_bytes = bytes32.from_hexstr(root_hash)
+        root_changed, answer = await self.data_store.answer_server_query(
+            node_hash_bytes, tree_id_bytes, root_hash_bytes
+        )
+        return web.json_response(
+            {
+                "root_changed": root_changed,
+                "answer": answer,
+            }
+        )
+
+    async def init_example_data_store(self) -> None:
```

Comment:
is this supposed to be production code?

Reply:
no, the production code should read the actual DataStore DB that other services write to. however I needed to start somewhere, and creating my own DataStore demonstrate the client/server functionality. will think of creating a test out of this.